src/Service.Tests/Unittests/SqlMetadataProviderUnitTests.cs

```diff
@@ -354,5 +355,121 @@ public void ValidateGraphQLReservedNaming_DatabaseColumns(string dbColumnName, s
                 actual: isViolationWithGraphQLGloballyDisabled,
                 message: "Unexpected failure. fieldName: " + dbColumnName + " | fieldMapping:" + mappedName);
         }
+
+        /// <summary>
+        /// Test to validate successful inference of relationship data based on data provided in the config and the metadata
+        /// collected from the MsSql database.
+        /// </summary>
+        [TestMethod, TestCategory(TestCategory.MSSQL)]
+        public async Task ValidateInferredRelationshipInfoForMsSql()
+        {
+            DatabaseEngine = TestCategory.MSSQL;
+            await SetupTestFixtureAndInferMetadata();
+            ValidateInferredRelationshipInfoForTables();
+        }
+
+        /// <summary>
+        /// Test to validate successful inference of relationship data based on data provided in the config and the metadata
+        /// collected from the MySql database.
+        /// </summary>
+        [TestMethod, TestCategory(TestCategory.MYSQL)]
```

Comment:
Optional: `FillInferredFkInfo()` and `GetValidatedFKs()` get executed inrrespective of the SQL database type. Having a test with a single SQL database type looks sufficient to exercise those code paths. Please let me know if I've missed something.